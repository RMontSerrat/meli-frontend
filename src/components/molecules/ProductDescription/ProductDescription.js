import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './ProductDescription.module.scss';

const ProductDescription = ({ description, classes }) => (
  <div className={classNames(styles.ProductDescription, classes.root)} data-testid="ProductDescription">
    <h2 className={classNames(styles.title, classes.title)}>
      Descrição do produto
    </h2>
    <p className={classNames(styles.content, classes.content)}>
      {description}
    </p>
  </div>
);

ProductDescription.propTypes = {
  description: PropTypes.node.isRequired,
  classes: PropTypes.shape({
    root: PropTypes.string,
    title: PropTypes.string,
    content: PropTypes.string,
  }),
};

ProductDescription.defaultProps = {
  classes: {},
};

export default ProductDescription;
