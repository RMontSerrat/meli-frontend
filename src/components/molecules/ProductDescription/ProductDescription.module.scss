.ProductDescription {
  .title {
    font-family: $font-family;
    font-size: size(xbig);
    font-weight: $font-regular;
    margin: 0;
    padding-bottom: 20px;
  }

  .content {
    color: color(dark, base);
    font-family: $font-family;
    font-size: size(medium);
    margin: 0;
  }
}
