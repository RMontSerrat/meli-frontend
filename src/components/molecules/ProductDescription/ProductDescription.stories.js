/* eslint-disable */
import React from 'react';
import { storiesOf } from '@storybook/react';
import ProductDescription from './ProductDescription';

storiesOf('ProductDescription', module).add('default', () => 
  <ProductDescription 
    description="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris interdum 
    gravida turpis ut auctor. Curabitur rhoncus fringilla ultrices. Aenean dapibus rutrum 
    mauris, in venenatis nunc ornare nec. Maecenas at dui vulputate neque pulvinar suscipit. 
    Mauris eros ipsum, aliquet vel laoreet nec, sollicitudin vel ante. Curabitur eu cursus turpis. Suspendisse"
  />);
