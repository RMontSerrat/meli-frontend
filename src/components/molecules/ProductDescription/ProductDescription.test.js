import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ProductDescription from './ProductDescription';

describe('<ProductDescription />', () => {
  test('it should mount', () => {
    render(<ProductDescription description="Description text" />);

    const productDescription = screen.getByTestId('ProductDescription');

    expect(productDescription).toBeInTheDocument();
  });

  test('snapshot', () => {
    const { asFragment } = render(
      <ProductDescription description="Description text" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render with classes', () => {
    render(
      <ProductDescription classes={{ root: 'rootClass', title: 'titleClass', content: 'contentClass' }} description="Description text" />,
    );
    const productDescription = screen.getByTestId('ProductDescription');
    const productDescriptionTitle = screen.getByTestId('ProductDescription-title');
    const productDescriptionContent = screen.getByTestId('ProductDescription-content');
    expect(productDescription.classList).toContain('rootClass');
    expect(productDescriptionTitle.classList).toContain('titleClass');
    expect(productDescriptionContent.classList).toContain('contentClass');
  });
});
