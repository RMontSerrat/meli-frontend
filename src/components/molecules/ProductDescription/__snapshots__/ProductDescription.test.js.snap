// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDescription /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="ProductDescription"
    data-testid="ProductDescription"
  >
    <h2
      class="title"
      data-testid="ProductDescription-title"
    >
      Descrição do produto
    </h2>
    <p
      class="content"
      data-testid="ProductDescription-content"
    >
      Description text
    </p>
  </div>
</DocumentFragment>
`;
